File: app/src/main/java/com/levibostian/androidblanky/db/models/SampleModel.java
package com.levibostian.androidblanky.db.models;

import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;
import io.realm.annotations.Required;

import java.util.Date;

public class SampleModel extends RealmObject {

    @PrimaryKey public long id;
    @Required public Date created;
    @Required public String title;
    public boolean done;
    public OtherSampleModel other_model;

    public SampleModel() {
    }

}

File: app/src/main/java/com/levibostian/androidblanky/db/models/SampleModel.kt
package com.levibostian.androidblanky.db.models

import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import io.realm.annotations.Required
import java.util.*

class SampleModel : RealmObject() {

    @PrimaryKey var id: Long = 0
    @Required var created: Date = Date()
    @Required var title: String = ""
    var done: Boolean = false
    var other_model: OtherSampleModel? = null

}